network/transport/grpc/connection_manager_test.go

```diff
@@ -128,12 +130,12 @@ func Test_grpcConnectionManager_acceptGRPCStream(t *testing.T) {
 	t.Run("new client", func(t *testing.T) {
 		cm := NewGRPCConnectionManager(Config{peerID: "server-peer-id"}).(*grpcConnectionManager)
 
-		serverStream := &stubServerStream{clientMetadata: constructMetadata("client-peer-id")}
+		serverStream := newServerStream("client-peer-id")
 		accepted, peerInfo, closer := cm.acceptGRPCStream(serverStream)
 
 		assert.True(t, accepted) // not already connected
 		assert.Equal(t, transport.PeerID("client-peer-id"), peerInfo.ID)
-		assert.Equal(t, "127.0.0.1", peerInfo.Address)
+		assert.Equal(t, "127.0.0.1:9522", peerInfo.Address)
```

Comment:
better to take a random available port?

Reply:
It doesn't actually open a port, because it's a stub gRPC stream. For the unit test the port is derived from the peer ID (some crc32 + `mod`-ing). Our existing integration tests, however, do take hardcoded ports. We might want to fix that at some point.